bind.sh

```diff
@@ -17,88 +17,131 @@
-            MEMS=(${2//\// })
-            if [[ "$IDX" -ge "${#MEMS[@]}" ]]; then
-                echo "Error: Incomplete MEM binding specification" 1>&2
-                exit 1
-            fi
-            ;;
-        --nics)
-            NICS=(${2//\// })
-            if [[ "$IDX" -ge "${#NICS[@]}" ]]; then
-                echo "Error: Incomplete NIC binding specification" 1>&2
-                exit 1
-            fi
-            ;;
-        *)
-            break
-            ;;
-    esac
-    shift 2
-done
+export LEGATE_RANK="$rank"
+
+if [ -n "${cpus+x}" ]; then
+  cpus=(${cpus//\// })
+  if [[ "$rank" -ge "${#cpus[@]}" ]]; then
+      echo "Error: Incomplete CPU binding specification" 1>&2
+      exit 1
+  fi
+fi
+
+if [ -n "${gpus+x}" ]; then
+  gpus=(${gpus//\// })
+  if [[ "$rank" -ge "${#gpus[@]}" ]]; then
+      echo "Error: Incomplete GPU binding specification" 1>&2
+      exit 1
+  fi
+  export CUDA_VISIBLE_DEVICES="${gpus[$rank]}"
+fi
 
-# Prepare environment
-if [[ -n "${GPUS+x}" ]]; then
-    export CUDA_VISIBLE_DEVICES="${GPUS[$IDX]}"
+if [ -n "${mems+x}" ]; then
+  mems=(${mems//\// })
+  if [[ "$rank" -ge "${#mems[@]}" ]]; then
+      echo "Error: Incomplete MEM binding specification" 1>&2
+      exit 1
+  fi
 fi
-if [[ -n "${NICS+x}" ]]; then
-    # Set all potentially relevant variables, hopefully they are ignored if we
-    # are not using the corresponding network.
-    NIC="${NICS[$IDX]}"
-    export UCX_NET_DEVICES="${NIC//,/:1,}":1
-    export NCCL_IB_HCA="$NIC"
-    NIC_ARR=(${NIC//,/ })
-    export GASNET_NUM_QPS="${#NIC_ARR[@]}"
-    export GASNET_IBV_PORTS="${NIC//,/+}"
+
+if [ -n "${nics+x}" ]; then
+  nics=(${nics//\// })
+  if [[ "$rank" -ge "${#nics[@]}" ]]; then
+      echo "Error: Incomplete NIC binding specification" 1>&2
+      exit 1
+  fi
+
+  # set all potentially relevant variables (hopefully they are ignored if we
+  # are not using the corresponding network)
+  nic="${nics[$rank]}"
+  nic_array=(${nic//,/ })
+  export UCX_NET_DEVICES="${nic//,/:1,}":1
+  export NCCL_IB_HCA="$nic"
+  export GASNET_NUM_QPS="${#nic_array[@]}"
+  export GASNET_IBV_PORTS="${nic//,/+}"
 fi
 
-# Prepare command
-if command -v numactl &> /dev/null; then
-    if [[ -n "${CPUS+x}" ]]; then
-        set -- --physcpubind "${CPUS[$IDX]}" "$@"
-    fi
-    if [[ -n "${MEMS+x}" ]]; then
-        set -- --membind "${MEMS[$IDX]}" "$@"
-    fi
-    set -- numactl "$@"
-elif [[ -n "${CPUS+x}" || -n "${MEMS+x}" ]]; then
-    echo "Warning: numactl is not available, cannot bind to cores or memories" 1>&2
+# numactl is only needed if cpu or memory pinning was requested
+if [[ -n "${cpus+x}" || -n "${mems+x}" ]]; then
+  if command -v numactl &> /dev/null; then
+      if [[ -n "${cpus+x}" ]]; then
+          set -- --physcpubind "${cpus[$rank]}" "$@"
+      fi
+      if [[ -n "${mems+x}" ]]; then
+          set -- --membind "${mems[$rank]}" "$@"
+      fi
+      set -- numactl "$@"
+  elif [[ -n "${cpus+x}" || -n "${mems+x}" ]]; then
```

Comment:
you already checked this above ;-)

Reply:
https://github.com/nv-legate/legate.core/pull/461/commits/6eef4c185b5038fad992352867009e85c7b34c56